src/App.js

```diff
@@ -1,5 +1,135 @@
+import { Console, Random } from '@woowacourse/mission-utils';
+
 class App {
-  async play() {}
+  constructor() {
+    this.numberLength = 3; // 숫자의 길이
+    this.computerNum = this.generateComputerNumber(); // 컴퓨터 숫자
+  }
+
+  async play() {
+    // 숫자 야구 게임의 메인 함수
+    Console.print('\n숫자 야구 게임을 시작합니다');
+
+    let isGameEnded = false;
+
+    while (!isGameEnded) {
+      const userNum = await this.getUserNumber(); // 사용자 숫자
+      if (userNum === null) {
+        Console.print('게임이 종료되었습니다.');
+        break;
+      }
+
+      const { strikes, balls } = this.compareNumbers(userNum); // 두 숫자 비교해서 strike와 ball 개수 세기
+      isGameEnded = await this.checkResult(strikes, balls);
+    }
+
+    return null;
+  }
+
+  generateComputerNumber() {
+    // 컴퓨터 숫자 랜덤으로 3자리 생성하기
+    const startInclusive = 1;
+    const endInclusive = 9;
+    const computerNumbers = new Set();
+
+    while (computerNumbers.size < this.numberLength) {
+      const number = Random.pickNumberInRange(startInclusive, endInclusive);
+      computerNumbers.add(number);
+    }
+
+    const computerNum = [...computerNumbers].join('');
+    return computerNum;
+  }
+
+  async getUserNumber() {
+    // 사용자로부터 3자리 숫자 입력받기
+    try {
+      const userInput = await Console.readLineAsync('3자리 숫자를 입력하세요 (1~9): ');
+      const userNum = userInput ? userInput.trim() : '';
+
+      if (userNum === null) {
+        throw new Error('[ERROR] 입력이 취소되었습니다.');
+      }
+
+      this.validateUserInput(userNum);
+      return userNum;
+    } catch (error) {
+      return Promise.reject(error); // 에러 처리 문구 주의하기!
+    }
+  }
+
+  validateUserInput(userNum) {
+    // 입력받은 값이 유효성 검사, 유효하지 않은 경우 throw로 에러 처리하기
+    if (userNum.length !== this.numberLength) {
+      throw new Error('[ERROR] 숫자는 3자리여야 합니다.');
+    }
+
+    if (!/^[1-9]{3}$/.test(userNum)) {
+      throw new Error('[ERROR] 1부터 9까지의 숫자로만 입력해야 합니다.');
+    }
+
+    const userNumbersSet = new Set(userNum.split(''));
+
+    if (this.numberLength !== userNumbersSet.size) {
+      throw new Error('[ERROR] 중복된 숫자를 입력하면 안됩니다.');
+    }
+  }
+
+  compareNumbers(userNum) {
+    // 컴퓨터 숫자와 사용자 숫자 비교하기
+    let strikes = 0;
+    let balls = 0;
+
+    const computerNumbersSet = new Set(this.computerNum.split(''));
+    const userNumbersSet = new Set(userNum.split(''));
+
+    for (const digit of userNumbersSet) {
+      if (computerNumbersSet.has(digit)) {
+        if (this.computerNum.indexOf(digit) === userNum.indexOf(digit)) {
+          // 숫자가 같고, 같은 자리에 있다면 strike
+          strikes++;
+        } else {
+          // 숫자가 같지만, 다른 자리에 있다면 ball
+          balls++;
+        }
+      }
+    }
```

Comment:
```suggestion
    for (const digit of userNumbersSet) {
      if (computerNumbersSet.has(digit)) {
        if (this.computerNum.indexOf(digit) === userNum.indexOf(digit)) {
          // 숫자가 같고, 같은 자리에 있다면 strike
          strikes++;
          continue;
        } 
        // 숫자가 같지만, 다른 자리에 있다면 ball
        balls++;
      }
    }
```

혹시 해당 방식을 사용하면 else를 작성하지 않아도 되지 않을까 생각했습니다.

Reply:
`continue`를 사용할 생각은 전혀 못 했는데 이 방법도 좋은 것 같아요!!

Reply:
앗! 그런데 eslint 규칙에 따르면, `continue`를 사용하면 코드 이해나 가독성이 떨어지기 때문에 권장하는 방법은 아니라고 합니다. 참고 문서 같이 드립니다~ 
[eslint continue docs](https://eslint.org/docs/latest/rules/no-continue)

Reply:
헉 그렇군요!! 감사합니다, 몰랐던 걸 알고가네요!